pkg/importer/dataStream_test.go

```diff
@@ -25,6 +46,78 @@ func createDataStream(ep, accKey, secKey string) *dataStream {
 	return ds
 }
 
+func createDataStreamMulti(testfile, ep string) *dataStream {
+	// set our directory for images
+	imageDir, _ := filepath.Abs("../../test/images")
+	urlPath := filepath.Join(ep, testfile)
+	testFilePath := filepath.Join(imageDir, testfile)
+
+	dsurl, _ := ParseEndpoint(urlPath)
+
+	ds := &dataStream{
+		Url:         dsurl,
+		buf:         make([]byte, image.MaxExpectedHdrSize),
+		accessKeyId: "",
+		secretKey:   "",
+	}
+
+	ds.constructReaders()
+
+	buf := bytes.NewBuffer(nil)
+	f, _ := os.Open(testFilePath)
+	io.Copy(buf, f)
+	f.Close()
+	testBytes, _ := ioutil.ReadAll(buf)
+
+	//append readers
+	xzreader, _ := xz.NewReader(bytes.NewReader(testBytes), 0)
+	tarreader := tar.NewReader(xzreader)
+
+	ds.appendReader(RdrXz, xzreader)
+	ds.appendReader(RdrTar, tarreader)
+
+	return ds
+}
+
+func createDataStreamBuf(ep, accKey, secKey, testfile string) (*dataStream, []byte, int) {
+
+	// set our directory for images
+	imageDir, _ := filepath.Abs("../../test/images")
+	localImageBase := filepath.Join("file://", imageDir)
+	urlPath := filepath.Join(localImageBase, testfile)
+	testFilePath := filepath.Join(imageDir, testfile)
+	if len(ep) == 0 {
+		ep = urlPath
+	}
+	dsurl, _ := ParseEndpoint(ep)
+
+	ds := &dataStream{
+		Url:         dsurl,
+		buf:         make([]byte, image.MaxExpectedHdrSize),
+		accessKeyId: accKey,
+		secretKey:   secKey,
+	}
+
+	ds.constructReaders()
+	// if no file supplied just return empty byte stream
+	if len(testfile) == 0 {
+		return ds, []byte{'T', 'E', 'S', 'T'}, 0
+	}
+
+	buf := bytes.NewBuffer(nil)
+	f, _ := os.Open(testFilePath)
+	fi, _ := f.Stat()
+	expectedSize := int(fi.Size())
+	io.Copy(buf, f)
+	f.Close()
+	testBytes, _ := ioutil.ReadAll(buf)
+
+	//append readers
+	ds.appendReader(RdrXz, bytes.NewReader(testBytes))
```

Comment:
just append the file reader `ds.appendReader(RdrFile, f)`

This isn't an xz reader anyway.